### PR TITLE
Add pagination permissions

### PR DESCRIPTION
**Pull request by mudkipdev**
This pull request will make sure nobody can interact with the `/tags list` menu other than the person who ran the command.

### CODE REVIEW

File: leaf/utils/pagination.py [hunk 1]
```diff
@@ -47,6 +57,16 @@ async def previous(
     async def page(
         self, interaction: discord.Interaction, button: discord.ui.Button
     ) -> None:
+        if self.author and interaction.user != self.author:
```
Comment (bsrecker) [inline leaf/utils/pagination.py:60]: Should we add some error handling here for cases where the interaction may not be valid anymore? Maybe surround with a try/except and log an exception or something.

File: leaf/utils/pagination.py [hunk 2]
```diff
@@ -57,6 +77,16 @@ async def next(
         await self.update(interaction)
 
     async def update(self, interaction: discord.Interaction) -> None:
+        if self.author and interaction.user != self.author:
```
Comment (bsrecker) [inline leaf/utils/pagination.py:80]: Same error handling here.

File: leaf/utils/pagination.py [hunk 3]
```diff
@@ -2,16 +2,23 @@
...
+        self,
+        embeds: List[discord.Embed],
+        *,
+        index: int = 0,
+        author: Optional[discord.User] = None
+    ) -> None:
+        self.embeds = embeds
+        self.index = index
+        self.author = author
+        self.paginated_view = PaginatedView(self.embeds, author=author)
```
Comment (bsrecker) [inline leaf/utils/pagination.py]: we should pass the variables to the PaginatedView class so they are not duplicated.

```
self.paginated_view = PaginatedView(
            embeds=self.embeds, index=self.index, author=self.author
        )
```

Then the following:

```
class PaginatedView(discord.ui.View):
    def __init__(
        self,
        *,
        embeds: List[discord.Embed],
        index: int,
        author: Optional[discord.User] = None
    ) -> None:
        super().__init__(timeout=None)
        self.embeds = embeds
        self.index = index
        self.author = author
```

**Review by bsrecker — CHANGES_REQUESTED** [review]
Suggested improvements.

File: leaf/utils/pagination.py (hunk 1 above)
Comment (mudkipdev) [reply leaf/utils/pagination.py:60]: @bsrecker In what case would it be invalid?

File: leaf/utils/pagination.py (hunk 3 above)
Comment (mudkipdev) [reply leaf/utils/pagination.py]: The index does not need to be passed as it is set using the `set_index` function.

File: leaf/utils/pagination.py (hunk 1 above)
Comment (bsrecker) [reply leaf/utils/pagination.py:60]: That I don't know. Theoretically it should never happen, but best practice is probably to proactively prevent the bot from crashing if it did happen.

**Pull request merged**